wire-gson-support/src/main/java/com/squareup/wire/MessageTypeAdapter.kt

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2013 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire
+
+import com.google.gson.Gson
+import com.google.gson.JsonElement
+import com.google.gson.TypeAdapter
+import com.google.gson.reflect.TypeToken
+import com.google.gson.stream.JsonReader
+import com.google.gson.stream.JsonToken
+import com.google.gson.stream.JsonWriter
+import com.squareup.wire.WireField.Label
+import java.io.IOException
+import java.math.BigInteger
+import java.util.ArrayList
+import java.util.Collections.unmodifiableMap
+import java.util.LinkedHashMap
+
+internal class MessageTypeAdapter<M : Message<M, B>, B : Message.Builder<M, B>>(
+  private val gson: Gson,
+  type: TypeToken<M>
+) : TypeAdapter<M>() {
+  private val messageAdapter: RuntimeMessageAdapter<M, B> =
+      RuntimeMessageAdapter.create(type.rawType as Class<M>)
+  private val fieldBindings: Map<String, FieldBinding<M, B>>
+
+  init {
+    val fieldBindings = LinkedHashMap<String, FieldBinding<M, B>>()
+    for (binding in messageAdapter.fieldBindings().values) {
```

Comment:
might be an `associate` method somewhere you could use

Reply:
Fixed